.github/workflows/build-tool-packages.yml

```diff
@@ -163,23 +163,34 @@ jobs:
         body: |
           ${{ steps.generate-release-body.outputs.RELEASE_BODY }}
 
+    - name: Generate hash for packages
+      run: |
+        $childItems = Get-Childitem -Path '.'
+        $childItems | Foreach-Object {
+          $packageObj = Get-Childitem -Path $_.FullName | Select-Object -First 1
+          Write-Host "Package: $($packageObj.Name)"
+          $actualHash = (Get-FileHash -Path $packageObj.FullName -Algorithm sha256).Hash
+          $hashString = "$actualHash $($packageObj.Name)"
+          Write-Host "$hashString"
+          Add-Content -Path ./hashes.sha256 -Value "$hashString"
+        } 
+
     - name: Upload release assets
       uses: actions/github-script@v6
       with:
         github-token: ${{ secrets.GITHUB_TOKEN }}
         script: |
           const fs = require('fs');
-
           for (let artifactDir of fs.readdirSync('.')) {
-            let artifactName = fs.readdirSync(`${artifactDir}`)[0];
+            let artifactName = fs.lstatSync(artifactDir).isDirectory() ? fs.readdirSync(`${artifactDir}`)[0] : artifactDir;
```

Comment:
How can we be certain that the first entry in `artifactDir` represents an `artifactName`? Can you please add comments explaining the use cases addressed by this branch? It is currently unclear why `artifactName` is determined in this manner. Adding a comment referencing the expected content of `.` and its subdirectories could be beneficial here.

Reply:
I think it is not necessary because the logic regarding to downloading and creation of file go in the previous steps. So it is not needed to describe it here.